src/miner.cpp

```diff
@@ -300,6 +300,10 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
             if (!ContextualCheckInputs(tx, state, view, true, MANDATORY_SCRIPT_VERIFY_FLAGS, true, Params().GetConsensus()))
                 continue;
 
+            // Check to make sure txs have not expired
+            if (IsExpiredTx(tx, nHeight))
```

Comment:
Put this check in the first loop, in the same conditional as `IsCoinbase()` and `!IsFinalTx()`. That way, we don't go to the trouble of reading into memory all the inputs of expired transactions.